handler.go

```diff
@@ -7,54 +7,55 @@ import (
 
 var ErrUnsupportedHandler = errors.New("siesta: unsupported handler")
 
-type contextHandler func(Context, http.ResponseWriter, *http.Request, func())
+// ContextHandler is a siesta handler.
+type ContextHandler func(Context, http.ResponseWriter, *http.Request, func())
 
-func (h contextHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	h(emptyContext{}, w, r, nil)
+func (h ContextHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	h(EmptyContext{}, w, r, nil)
 }
 
-func (h contextHandler) ServeHTTPInContext(c Context, w http.ResponseWriter, r *http.Request) {
+func (h ContextHandler) ServeHTTPInContext(c Context, w http.ResponseWriter, r *http.Request) {
 	h(c, w, r, nil)
 }
 
-func toContextHandler(f interface{}) contextHandler {
-	var m contextHandler
-
+// ToContextHandler transforms f into a ContextHandler.
+// f must be a function with one of the following signatures:
+//     func(http.ResponseWriter, *http.Request)
+//     func(http.ResponseWriter, *http.Request, func())
+//     func(Context, http.ResponseWriter, *http.Request)
+//     func(Context, http.ResponseWriter, *http.Request, func())
+func ToContextHandler(f interface{}) ContextHandler {
 	switch f.(type) {
 	case func(Context, http.ResponseWriter, *http.Request, func()):
-		m = contextHandler(f.(func(Context, http.ResponseWriter, *http.Request, func())))
-	case contextHandler:
-		m = f.(contextHandler)
+		return ContextHandler(f.(func(Context, http.ResponseWriter, *http.Request, func())))
```

Comment:
idiomatic `switch t := f.(type) {`?  That'd simplify most returns, this one would be `return ContextHandler(t)`